src/components/Datepicker/Datepicker.js

```diff
@@ -31,97 +32,137 @@ export class Datepicker extends React.Component {
   constructor(props, context) {
     super(props, context);
     this.prefix = (classes, passThrough) => prefixClasses(this.context.cssPrefix, classes, passThrough);
-    const { open, preselectedDate } = this.props;
+    const { date, locale, open, timezone, translations } = this.props;
+
+    moment.locale(locale.toLowerCase());
+    moment.tz.setDefault(timezone);
+    const isValidDate = moment(date, iso8601DateFormat, true).isValid();
 
     this.state = {
       open,
-      preselectedDate: preselectedDate ? moment(preselectedDate) : null,
-      inputDate: preselectedDate ? moment(preselectedDate).format(salesforceDateFormat).toString() : '',
-      viewedDate: moment(),
+      inputValue: isValidDate ? moment(date).format(defaultDateFormat).toString() : '',
+      viewedDate: isValidDate ? moment(date) : moment(),
+    };
+
+    this.mergedTranslations = {
+      ...Datepicker.defaultProps.translations,
+      ...translations,
     };
   }
 
-  /**
-   * Callback when the year changes
-   * @param  {Object} e <select /> onChange event
-   * @return {void}     interacts with component state
-   */
-  onYearChange = (e) => {
-    const { viewedDate } = this.state;
+  componentWillReceiveProps(nextProps) {
+    const { onChange } = this.props;
+    const { date } = nextProps;
+    const nextDate = moment(date);
 
-    this.setState({ viewedDate: viewedDate.set('year', e.target.value) });
-  };
+    const isValid = moment(date, iso8601DateFormat, true).isValid();
 
-  /**
-   * Callback when the month changes
-   * @param  {Number} relativeMonth one month forward or backward
-   * @return {void}                 interacts with component state
-   */
-  onMonthChange = (relativeMonth) => {
-    const { viewedDate } = this.state;
+    const output = isValid ? nextDate.format(iso8601DateFormat) : null;
+    onChange(output, isValid);
```

Comment:
This will fire every time a prop is changed, whether it was an actual date change or not.  I'd specifically add a string comparison between `nextProps.date` and the current `date` to see if the `onChange()` call is necessary.